test/f32-conv-hwc.cc

```diff
@@ -2,5047 +2,877 @@
+    }                                                                                                                  \
+  }
+
+#define XNN_TEST_CONVHWC_OUTPUT_Y_START(                                                                               \
+  ukernel, arch_flags, kernel_sizes, subsamplings, padding_right_, padding_left_, input_channel, output_channel_tile,  \
+  input_widths, init_params, ...)                                                                                      \
+  TEST(ukernel, output_y_start)                                                                                        \
+  {                                                                                                                    \
+    for (size_t output_y_starts = 1; output_y_starts <= 3; output_y_starts++) {                                        \
+      for (size_t output_channels = 1; output_channels < output_channel_tile * 2;                                      \
+           output_channels += output_channel_tile - 1) {                                                               \
+        for (size_t input_widths_ = (padding_left_ ? 1 : 2); input_widths_ < input_widths * 8;                         \
+             input_widths_ += (input_widths * 2 - 1)) {                                                                \
+          auto tester = ConvHWCMicrokernelTester()                                                                     \
+                          .kernel_size(kernel_sizes)                                                                   \
+                          .subsampling(subsamplings)                                                                   \
+                          .input_channels(input_channel)                                                               \
+                          .output_channels_tile(output_channel_tile)                                                   \
+                          .output_channels(output_channels)                                                            \
+                          .input_width(input_widths_)                                                                  \
+                          .input_height(9)                                                                             \
+                          .output_y_start(output_y_starts);                                                            \
+          if (padding_left_ == 1 && padding_right_ == 1) {                                                             \
+            tester.padding_width(padding_right_);                                                                      \
+          }                                                                                                            \
+          else if (padding_left_ == 0 && padding_right_ == 1) {                                                        \
```

Comment:
Is this condition right? It looks weird, the body of both ifs is the same. Why not `if ((padding_left_ == 0 || padding_left_ == 1) && padding_right_ == 1)` ?

Reply:
Yes, the condition is correct. For (padding_left_ == 1 && padding_right_ == 1), we use **padding_width**(padding_right_), and for (padding_left_ == 0 && padding_right_ == 1), we use **padding_right**(padding_right_).

Reply:
I see, I missed the difference